c_binding/libsg.c

```diff
@@ -803,3 +813,130 @@ int _sg_tp_sas_addr_of_disk(char *err_msg, int fd, char *tp_sas_addr)
 
     return rc;
 }
+
+int _sg_io_mode_sense(char *err_msg, int fd, uint8_t page_code,
+                      uint8_t sub_page_code, uint8_t *data)
+{
+    int rc = LSM_ERR_OK;
+    uint8_t tmp_data[_SG_T10_SPC_MODE_SENSE_MAX_LEN];
+    uint8_t cdb[_T10_SPC_MODE_SENSE_CMD_LEN];
+    uint8_t sense_data[_T10_SPC_SENSE_DATA_MAX_LENGTH];
+    int ioctl_errno = 0;
+    char strerr_buff[_LSM_ERR_MSG_LEN];
+    uint8_t sense_key = _T10_SPC_SENSE_KEY_NO_SENSE;
+    char sense_err_msg[_LSM_ERR_MSG_LEN];
+    struct _sg_t10_mode_para_hdr *mode_hdr = NULL;
+    uint16_t block_dp_len = 0;
+    uint16_t mode_data_len = 0;
+
+    assert(err_msg != NULL);
+    assert(fd >= 0);
+    assert(data != NULL);
+
+    memset(sense_err_msg, 0, _LSM_ERR_MSG_LEN);
+    memset(data, 0, _SG_T10_SPC_MODE_SENSE_MAX_LEN);
+
+    /* SPC-5 Table 171 — MODE SENSE(10) command */
+    cdb[0] = MODE_SENSE_10;                     /* OPERATION CODE */
+    cdb[1] = 0;                                 /* disable block descriptors
+                                                 * and long LBA accepted
+                                                 */
+    /* We don't need block descriptors or long LBA accepted */;
+    cdb[2] = page_code & UINT8_MAX;             /* PAGE CODE and Page control*/
+    /* The page control is 0 means 'current values' */
+    cdb[3] = sub_page_code & UINT8_MAX;         /* Subpage code */
+    cdb[4] = 0;                                 /* Reserved */
+    cdb[5] = 0;                                 /* Reserved */
+    cdb[6] = 0;                                 /* Reserved */
+    cdb[7] = (_SG_T10_SPC_MODE_SENSE_MAX_LEN >> 8 )& UINT8_MAX;
```

Comment:
Is there a particular reason for the space after the 8 and no space between the right parenthesis and the &?